### PR TITLE
Added get case endpoint to retrieve case from cms

### PR DESCRIPTION
**Pull request by nitinprabhu**
- https://tools.hmcts.net/jira/browse/NFDIV-436

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/divorce/orchestration/controller/advice/GlobalExceptionHandler.java [hunk 1]
```diff
@@ -59,6 +59,13 @@
             );
     }
 
+    @ExceptionHandler(CaseNotFoundException.class)
+    ResponseEntity<Object> handleCaseNotFoundException(CaseNotFoundException exception) {
+        log.warn(exception.getMessage(), exception);
+
+        return ResponseEntity.status(HttpStatus.NOT_FOUND).body(exception.getMessage());
+    }
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/reform/divorce/orchestration/controller/advice/GlobalExceptionHandler.java]: Coverage of this is shown as 0 in sonar but it is actually covered by integration test

File: src/main/java/uk/gov/hmcts/reform/divorce/orchestration/service/impl/CaseServiceImpl.java [hunk 2]
```diff
@@ -31,4 +38,23 @@
         }
     }
 
+    @Override
+    public CaseDataResponse getCase(String authorizationToken) throws CaseNotFoundException {
+        CaseDetails caseDetails = caseMaintenanceClient.getCaseFromCcd(authorizationToken);
+
+        if (caseDetails == null) {
```
Comment (mapingo) [inline src/main/java/uk/gov/hmcts/reform/divorce/orchestration/service/impl/CaseServiceImpl.java]: Do we need to test for null here?  Won't CMS have returned a 404 if there is no case?

**Review by mapingo — APPROVED** [review]

**Review by SarahLittlejohn — APPROVED** [review]
